src/sentry/integrations/jira/utils/create_issue_schema_transformers.py

```diff
@@ -0,0 +1,133 @@
+from collections.abc import Callable, Iterable, Mapping
+from typing import Any, TypeVar
+
+from sentry.integrations.jira.models.create_issue_metadata import (
+    JIRA_CUSTOM_FIELD_TYPES,
+    JiraField,
+    JiraSchemaTypes,
+)
+from sentry.shared_integrations.exceptions import IntegrationFormError
+
+
+class JiraSchemaParseError(Exception):
+    pass
+
+
+def parse_number_field(num_str: Any) -> int | float:
+    try:
+        if isinstance(num_str, str) and "." in num_str:
+            return float(num_str)
+
+        return int(num_str)
+    except ValueError:
+        raise JiraSchemaParseError(f"Invalid number value provided for field: '{num_str}'")
```

Comment:
This is only used here for now, but I imagine we can start adding stricter validation with explicit error raises when we want to tighten up our inputs a bit.